### PR TITLE
DDCE-5224 - Play 3 Migration

### PR DESCRIPTION
**Pull request by machinesleet**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+ThisBuild / majorVersion := 0
+
+lazy val microservice = (project in file("."))
   .enablePlugins(PlayScala, SbtDistributablesPlugin)
   .disablePlugins(JUnitXmlReportPlugin) //Required to prevent https://github.com/scalatest/scalatest/issues/1427
-  .configs(IntegrationTest)
-  .settings(inConfig(IntegrationTest)(itSettings))
-  .settings(inConfig(Test)(testSettings))
   .settings(
     DefaultBuildSettings.scalaSettings,
```
Comment (frikit) [inline build.sbt:17]: delete please

File: build.sbt [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
     DefaultBuildSettings.defaultSettings(),
```
Comment (frikit) [inline build.sbt:18]: delete please

File: build.sbt [hunk 3]
```diff
@@ -1,21 +1,21 @@
 import play.sbt.routes.RoutesKeys
-import sbt.Def
 import scoverage.ScoverageKeys
 import uk.gov.hmrc.DefaultBuildSettings
 import uk.gov.hmrc.versioning.SbtGitVersioning.autoImport.majorVersion
```
Comment (frikit) [inline build.sbt:4]: delete please

File: build.sbt [hunk 4]
```diff
@@ -1,21 +1,21 @@
 import play.sbt.routes.RoutesKeys
-import sbt.Def
 import scoverage.ScoverageKeys
 import uk.gov.hmrc.DefaultBuildSettings
```
Comment (frikit) [inline build.sbt:3]: delete please

File: build.sbt [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
   .enablePlugins(PlayScala, SbtDistributablesPlugin)
   .disablePlugins(JUnitXmlReportPlugin) //Required to prevent https://github.com/scalatest/scalatest/issues/1427
-  .configs(IntegrationTest)
-  .settings(inConfig(IntegrationTest)(itSettings))
-  .settings(inConfig(Test)(testSettings))
   .settings(
     DefaultBuildSettings.scalaSettings,
     DefaultBuildSettings.defaultSettings(),
-    inConfig(Test)(testSettings),
     majorVersion := 0,
```
Comment (frikit) [inline build.sbt:19]: delete please

File: build.sbt [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
     scalaVersion := "2.13.12",
```
Comment (frikit) [inline build.sbt:20]: delete please

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -2,28 +2,22 @@ import sbt.*
 
 object AppDependencies {
 
-  val boostrapVersion = "7.22.0"
-  val mongoVersion = "1.3.0"
+  private val bootstrapVersion = "8.5.0"
```
Comment (frikit) [inline project/AppDependencies.scala:5]: update to 8.6.0 and plugin

File: project/plugins.sbt [hunk 8]
```diff
@@ -1,17 +1,11 @@
...
-  "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
-)
-
-addSbtPlugin("uk.gov.hmrc"          % "sbt-auto-build"          % "3.20.0")
+addSbtPlugin("uk.gov.hmrc"          % "sbt-auto-build"          % "3.21.0")
 addSbtPlugin("uk.gov.hmrc"          % "sbt-distributables"      % "2.5.0")
-addSbtPlugin("com.typesafe.play"    % "sbt-plugin"              % "2.8.20")
-addSbtPlugin("org.scalastyle"       %% "scalastyle-sbt-plugin"  % "1.0.0")
-addSbtPlugin("org.scoverage"        % "sbt-scoverage"           % "2.0.9")
+addSbtPlugin("org.playframework"    % "sbt-plugin"              % "3.0.2")
```
Comment (frikit) [inline project/plugins.sbt:6]: update please

File: test/base/SpecBase.scala [hunk 9]
```diff
@@ -46,11 +46,15 @@ trait SpecBase extends PlaySpec with GuiceOneAppPerSuite with Mocked with TryVal
 
   implicit def messages: Messages = messagesApi.preferred(fakeRequest)
 
+  val defaultAppConfig = Seq(
```
Comment (frikit) [inline test/base/SpecBase.scala:49]: please add these 2 props:
```scala
 "auditing.enabled" -> false,
    "metrics.enabled" -> false,
```

**Review by frikit — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala (hunk 7 above)
Comment (machinesleet) [reply project/AppDependencies.scala]: which plugin?

**Review by frikit — APPROVED** [review]

**Pull request merged**